sky/jobs/server/core.py

```diff
@@ -282,12 +303,44 @@ def launch(
                 # workspace A, but the controller is in workspace B, the
                 # intermediate bucket and newly created bucket should be in
                 # workspace A.
-                return execution.launch(task=controller_task,
-                                        cluster_name=controller_name,
-                                        stream_logs=stream_logs,
-                                        retry_until_up=True,
-                                        fast=True,
-                                        _disable_controller_check=True)
+                if consolidation_mode_job_id is None:
+                    return execution.launch(task=controller_task,
+                                            cluster_name=controller_name,
+                                            stream_logs=stream_logs,
+                                            retry_until_up=True,
+                                            fast=True,
+                                            _disable_controller_check=True)
+                # Manually launch the scheduler process in consolidation mode.
+                local_handle = backend_utils.is_controller_accessible(
+                    controller=controller, stopped_message='')
+                backend = backend_utils.get_backend_from_handle(local_handle)
+                assert isinstance(backend, backends.CloudVmRayBackend)
+                backend.sync_file_mounts(
+                    handle=local_handle,
+                    all_file_mounts=controller_task.file_mounts,
+                    storage_mounts=controller_task.storage_mounts)
```

Comment:
Wait, why do we need to do this? Can't we just rsync directly to the destination cluster and avoid permuting the file mounts at all? That is, skip this whole section https://github.com/skypilot-org/skypilot/blob/660d2bde857362e597fa3c30252693fe4d8c3939/sky/jobs/server/core.py#L150-L177